src/transformers/processing_utils.py

```diff
@@ -85,6 +99,77 @@ def __init__(self, *args, **kwargs):
 
             setattr(self, attribute_name, arg)
 
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serializes this instance to a Python dictionary.
+
+        Returns:
+            `Dict[str, Any]`: Dictionary of all the attributes that make up this processor instance.
+        """
+        output = copy.deepcopy(self.__dict__)
+
+        # Get the kwargs in `__init__`.
+        sig = inspect.signature(self.__init__)
+        # Only save the attributes that are presented in the kwargs of `__init__`.
+        output = {k: v for k, v in output.items() if k in sig.parameters and k not in self.__class__.attributes}
```

Comment:
Would like to hear from @amyeroberts  for this part as they have the experience for `ImageProcessingMixin`.

#### Question in short:

Is it OK we only save the attributes that are also presented in `__init__`'s argument?

#### Details 

Basically, for `ImageProcessingMixin`, it seems we have the correspondence between `the attributes of self` and the arguments presented in `self.__init__`.

However, for `Processor`, this is not the case.

Example 1:

```python
class FuyuProcessor(ProcessorMixin):

    def __init__(self, image_processor, tokenizer):
        ...
        ...
        self.max_tokens_to_generate = 10
        self.max_position_embeddings = 16384
```
For `FuyuProcessor`, we could add those attributes to `__init__` arguments.

Example 2:

```python
class Kosmos2Processor(ProcessorMixin):

    def __init__(self, image_processor, tokenizer, num_patch_index_tokens=1024):
        tokenizer.return_token_type_ids = False

        self.eod_token = "</doc>"
        self.boi_token = "<image>"
        self.tag_tokens = [....]

        self.num_patch_index_tokens = num_patch_index_tokens
```
Some of the (token) attributes have to be there (used in other methods) and others are kept as information. But **it doesn't make much sense to put them in the arguments as their values are fixed**.

Reply:
I think it's better to just save out what the user can modify themselves i.e. what's in the `__init__`. If I looked at a config file e.g.:

```json
{
    "a": 1,
    "b": 2,
    "c": 3,
}
```

I would assume that I could modify `a`, `b` and `c`. However if e.g. `c` is hard-coded in the config then doing:

```py
config = FooConfig.from_pretrained(checkpoint, c=6)
```

Would result in unexpected behaviour and the only way for users to know what they can modify is looking into the code which is something we want to avoid.